integration/auth/auth_test.go

```diff
@@ -53,12 +61,423 @@ func TestMain(m *testing.M) {
+	if err != nil {
+		t.Error(err)
+	}
+	if u.Disabled {
+		t.Errorf("expecting user enabled")
+	}
+}
+
+func testUpdateUser(t *testing.T) {
+	u, err := client.GetUser(context.Background(), testFixtures.sampleUserBlank.UID)
+
+	if err != nil || u == nil {
+		t.Errorf("error getting user %s", err)
+	}
+	refU := &auth.UserRecord{
+		UserInfo: &auth.UserInfo{UID: testFixtures.sampleUserBlank.UID},
+		UserMetadata: &auth.UserMetadata{
+			CreationTimestamp: testFixtures.sampleUserBlank.UserMetadata.CreationTimestamp,
+		},
+	}
+	if !reflect.DeepEqual(u, refU) {
+		t.Errorf("\ngot %s, \nexpecting %s", toString(refU), toString(u))
+	}
+	utup := (&auth.UserToUpdate{}).
+		Disabled(false).
+		DisplayName("name").
+		PhoneNumber("+12345678901").
+		PhotoURL("http://photo.png").
+		Email("abc@ab.ab").
+		EmailVerified(true).
+		Password("wordpass").
+		CustomClaims(map[string]interface{}{"custom": "claims"})
+
+	_, err = client.UpdateUser(context.Background(), u.UID, utup)
+	if err != nil {
+		t.Error(err)
+	}
+
+	refU = &auth.UserRecord{
+		UserInfo: &auth.UserInfo{
+			UID:         testFixtures.sampleUserBlank.UID,
+			DisplayName: "name",
+			PhoneNumber: "+12345678901",
+			PhotoURL:    "http://photo.png",
+			Email:       "abc@ab.ab",
+		},
+		UserMetadata: &auth.UserMetadata{
+			CreationTimestamp: testFixtures.sampleUserBlank.UserMetadata.CreationTimestamp,
+		},
+		Disabled:      false,
+		EmailVerified: true,
+		CustomClaims:  map[string]interface{}{"custom": "claims"},
+	}
+	u, err = client.GetUser(context.Background(), u.UID)
+	testPI(u.ProviderUserInfo,
+		&auth.UserInfo{
+			DisplayName: "name",
+			Email:       "abc@ab.ab",
+			PhotoURL:    "http://photo.png",
+			ProviderID:  "password"},
+		&auth.UserInfo{
+			PhoneNumber: "+12345678901",
+			ProviderID:  "phone"},
+		t)
+	u.ProviderUserInfo = nil
+	if !reflect.DeepEqual(u, refU) {
+		t.Errorf("\ngot %s\nexpecting %s", toString(u), toString(refU))
+	}
+}
+
+func testPI(pi []*auth.UserInfo, passwordUI, phoneUI *auth.UserInfo, t *testing.T) {
+	var compareWith *auth.UserInfo
+	for _, ui := range pi {
+		switch ui.ProviderID {
+		case "password":
+			compareWith = passwordUI
+		case "phone":
+			compareWith = phoneUI
+		}
+		if !reflect.DeepEqual(ui, compareWith) {
+			t.Errorf("\ngot %#v, \nexpecting %#v", ui, compareWith)
+		}
+	}
+}
+
+func testRemoveDisplayName(t *testing.T) {
+	u, err := client.GetUser(context.Background(), testFixtures.sampleUserBlank.UID)
+	if err != nil {
+		t.Error(err)
+	}
+	if u.DisplayName == "" {
+		t.Errorf("expecting non empty display name")
+	}
+
+	_, err = client.UpdateUser(context.Background(), u.UID, (&auth.UserToUpdate{}).DisplayName(""))
+	if err != nil {
+		t.Error(err)
+	}
+	u, err = client.GetUser(context.Background(), testFixtures.sampleUserBlank.UID)
+	if u.DisplayName != "" {
+		t.Errorf("expecting non empty display name")
+	}
+}
+
+func testRemovePhotoURL(t *testing.T) {
```

Comment:
Combine these 3 remove test cases into a single test case, and make one UpdateUser call.